subprojects/core/src/main/java/org/gradle/execution/ProjectExecutionServices.java

```diff
@@ -104,38 +106,44 @@ EmptySourceTaskSkipper createEmptySourceTaskSkipper(
-                                    PropertyWalker propertyWalker,
-                                    TaskExecutionGraphInternal taskExecutionGraph,
-                                    TaskExecutionListener taskExecutionListener,
-                                    TaskListenerInternal taskListenerInternal,
-                                    TaskCacheabilityResolver taskCacheabilityResolver,
-                                    WorkExecutor<ExecutionRequestContext, CachingResult> workExecutor,
-                                    Deleter deleter,
-                                    ListenerManager listenerManager,
-                                    ReservedFileSystemLocationRegistry reservedFileSystemLocationRegistry,
-                                    EmptySourceTaskSkipper emptySourceTaskSkipper
+    TaskExecuter createTaskExecuter(
+        AsyncWorkTracker asyncWorkTracker,
+        BuildCacheController buildCacheController,
+        BuildOperationExecutor buildOperationExecutor,
+        BuildOutputCleanupRegistry cleanupRegistry,
+        BuildScanPluginApplied buildScanPlugin,
+        ClassLoaderHierarchyHasher classLoaderHierarchyHasher,
+        Deleter deleter,
+        EmptySourceTaskSkipper emptySourceTaskSkipper,
+        ExecutionHistoryStore executionHistoryStore,
+        FileCollectionFactory fileCollectionFactory,
+        FileCollectionFingerprinterRegistry fingerprinterRegistry,
+        ListenerManager listenerManager,
+        OutputChangeListener outputChangeListener,
+        OutputFilesRepository outputFilesRepository,
+        PropertyWalker propertyWalker,
+        ReservedFileSystemLocationRegistry reservedFileSystemLocationRegistry,
+        StartParameter startParameter,
+        TaskActionListener actionListener,
+        TaskCacheabilityResolver taskCacheabilityResolver,
+        TaskExecutionGraphInternal taskExecutionGraph,
+        TaskExecutionListener taskExecutionListener,
+        TaskExecutionModeResolver repository,
+        TaskListenerInternal taskListenerInternal,
+        TaskSnapshotter taskSnapshotter,
+        WorkExecutor<ExecutionRequestContext, CachingResult> workExecutor
     ) {
 
-        boolean buildCacheEnabled = buildCacheController.isEnabled();
-        boolean scanPluginApplied = buildScanPlugin.isBuildScanPluginApplied();
-
         TaskExecuter executer = new ExecuteActionsTaskExecuter(
-            buildCacheEnabled,
-            scanPluginApplied,
+            buildCacheController.isEnabled()
+                ? ExecuteActionsTaskExecuter.BuildCacheState.ENABLED
+                : ExecuteActionsTaskExecuter.BuildCacheState.DISABLED,
+            buildScanPlugin.isBuildScanPluginApplied()
+                ? ExecuteActionsTaskExecuter.ScanPluginState.APPLIED
+                : ExecuteActionsTaskExecuter.ScanPluginState.NOT_APPLIED,
+            VirtualFileSystemServices.isPartialInvalidationEnabled(startParameter.getSystemPropertiesArgs())
```

Comment:
Do the start parameter system property args contain the ones from `gradle.properties` as well?

Reply:
No it doesn't. :/

Reply:
Interestingly, IIUC this is what happens with instant execution: you cannot enable it via `systemProp.org.gradle...` in `gradle.propertie`. :/

https://github.com/gradle/gradle/blob/744d8995ddaf96bf14cb3053b4cb1dc69259e140/subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionHost.kt#L82-L83

I suggest we leave it as is, and fix it via `BuildOption`s instead.